File: leet-code/shuffle.go
package main

func shuffle(nums []int, n int) []int {
	m, result := 0, make([]int, len(nums))

	for i := range nums {
		if i%2 == 0 {
			result[i] = nums[m]
			m++
		} else {
			result[i] = nums[n]
			n++
		}
	}

	return result
}

File: leet-code/shuffle.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @return {number[]}
 */
var shuffle = function (nums, n) {
  let m = 0
  let result = []
  for (let i = 0; i < nums.length; i++) {
    if (i % 2 === 0) {
      result[i] = nums[m++]
    } else {
      result[i] = nums[n++]
    }
  }
  return result
}
const arr = [1, 3, 5, 2, 4, 6]
console.log(shuffle(arr, 3))

